lib/Backend/arm64/LowerMD.cpp

```diff
@@ -2086,28 +2086,19 @@ LowererMD::ChangeToAssign(IR::Instr * instr, IRType destType)
             instr->ReplaceSrc1(instr->GetSrc1()->UseWithNewType(TyFloat64, instr->m_func));
         }
     }
-    else if (TySize[destType] > TySize[srcType] && (IRType_IsSignedInt(destType) || IRType_IsUnsignedInt(destType)))
+    else if (!src->IsIndirOpnd() && TySize[destType] > TySize[srcType] && (IRType_IsSignedInt(destType) || IRType_IsUnsignedInt(destType)))
```

Comment:
>IsIndirOpnd [](start = 19, length = 11)

Just confirm that indir opnd won't have a size mismatch as we will hoist it and then again call LegalizeInstr